### ISSUE TITLE
It's COREWEBVIEW2_WEB_RESOURCE_CONTEXT_FETCH working?

### ISSUE THREAD

**Issue by dec** [MicrosoftEdge/WebViewFeedback#444]
Hello to all,

First of all, thanks very, very much, for the great work you all do in the WebView2. I decide to post this because I can't find any issue related with "COREWEBVIEW2_WEB_RESOURCE_CONTEXT_FETCH". I post this as a question and not as a bug since I am not sure if finally I am doing something wrong.

What I am trying to do is to handle "COREWEBVIEW2_WEB_RESOURCE_CONTEXT_FETCH" for local files requests using the "WebResourceRequested" event, however, when I try to use the Javascript "fetch" with a local file, the browser's console says that "fetch" it's not available for the file protocol, only for "http" and "https".

Is there a way to allow to use "fetch" with local files?

In other words... what "COREWEBVIEW2_WEB_RESOURCE_CONTEXT_FETCH" is expected to do? Because I mention that the browser's console output the referred error... but there is the fact that the "WebResourceRequested" event is apparently not executed when I use "fetch"... so what I am missing?

Thanks very much in advance for your help!

**Comment by ukandrewc** [comment]
To load local files, you'd need to use FileReader.

**Comment by dec** [comment]
Hello to all,

Thanks for your reply @ukandrewc!  However, supose I want to use "fetch"... just because some posible apps who run in the WebView uses it... not because I wanted to use it, for example...

My point here is why "COREWEBVIEW2_WEB_RESOURCE_CONTEXT_FETCH" is not takin in consideration (if that's the case), in the same way than "COREWEBVIEW2_WEB_RESOURCE_CONTEXT_XML_HTTP_REQUEST" is considered? Because this last one works well also for local files...

**Comment by ukandrewc** [comment]
Because XHR can access local files (although I'm suprised it still works because a Chrome security update last year, was supposed to block file access) but Fetch cannot - the Fetch spec doesn't include File protocols..

**Comment by dec** [comment]
Hello to all,

Thanks for your reply @ukandrewc. Well. Certainly I try right now with "fetch" and "http" and certainly the "WebResourceRequested" event is fired as expected... so in conclusion we can apparently handle "fetch" calls to "http" or "https", but not for "file" (local files".

That's what I can undertand... and it's right, don't you?

**Comment by ukandrewc** [comment]
That's right.

**Comment by dec** [comment]
Hello to all,

Thanks very much for your help @ukandrewc. :-)

**Comment by champnic** [comment]
Confirming what @ukandrewc said. Thanks everyone!
